core/http/src/main/java/org/trellisldp/http/impl/PutHandler.java

```diff
@@ -139,7 +143,11 @@ public ResponseBuilder initialize(final Resource parent, final Resource resource
             throw new NotAcceptableException();
         }
 
-        ofNullable(resource).flatMap(Resource::getContainer).ifPresent(p -> setParent(parent));
+        if (createUncontained) {
+            ofNullable(resource).flatMap(Resource::getContainer).ifPresent(p -> setParent(parent));
```

Comment:
Can you leave an inline comment here for this? It's not obvious to me quite what is happening…

Reply:
Definitely. There are a few more things to clean up from this PR, which I'll add in another commit